pkg/integration/tests/branch/sort_local_branches.go

```diff
@@ -0,0 +1,48 @@
+package branch
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var SortLocalBranches = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Sort local branches alphabetically or by recency",
+	ExtraCmdArgs: []string{},
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
+	SetupRepo: func(shell *Shell) {
+		shell.NewBranch("first")
+		shell.EmptyCommitWithDate("commit", "2023-04-07 10:00:00")
+		shell.NewBranch("second")
+		shell.EmptyCommitWithDate("commit", "2023-04-07 12:00:00")
+		shell.NewBranch("third")
+		shell.EmptyCommitWithDate("commit", "2023-04-07 11:00:00")
+		shell.Checkout("first")
+	},
+	Run: func(t *TestDriver, keys config.KeybindingConfig) {
+		// sorted by recency by default
+		t.Views().Branches().
+			Focus().
+			Lines(
+				Contains("first").IsSelected(),
+				Contains("third"),
+				Contains("second"),
+			).
+			SelectNextItem() // to test that the selection jumps back to the first when sorting
+
+		t.Views().Branches().
+			Press(keys.Branches.SortOrder)
+
+		t.ExpectPopup().Menu().Title(Equals("Sort order")).
+			Select(Contains("-committerdate")).
+			Confirm()
+
+		t.Views().Branches().
+			IsFocused().
+			Lines(
+				Contains("first").IsSelected(),
+				Contains("second"),
+				Contains("third"),
+			)
```

Comment:
Would be nice to also switch to alphabetical. This requires assigning the committer dates differently so that the order is actually different.

Reply:
Fixed in https://github.com/jesseduffield/lazygit/pull/3182/commits/de3d48f98f7121be1f073c6ba656a3b6bf85d60f. I've added a new "master" branch that stays selected at the top, instead of modifying the commit dates.